### PR TITLE
Add Service Bus queue definition to terraform

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-647

### Change description ###

Add Service Bus queue definition to terraform

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -7,7 +7,38 @@ resource "azurerm_resource_group" "rg" {
...
+  tenant_id               = "${var.tenant_id}"
+  object_id               = "${var.jenkins_AAD_objectId}"
+  resource_group_name     = "${local.vaultResourceGroupName}"
+  # dcd_cc_dev group object ID
+  product_group_object_id = "38f9dea6-e861-4a50-9e73-21e64f563537"
+}
+
+module "servicebus-namespace" {
+  source              = "git@github.com:hmcts/terraform-module-servicebus-namespace.git"
+  name                = "${var.product}-${var.component}-servicebus-${var.env}"
```
Comment (timja) [inline infrastructure/main.tf]: could we remove service bus from the name?
https://tools.hmcts.net/confluence/display/CNP/Naming+Conventions

File: infrastructure/main.tf [hunk 2]
```diff
@@ -26,3 +57,9 @@ module "bulk-scan-orchestrator" {
     LOGBACK_REQUIRE_ERROR_CODE  = false
   }
 }
+
+resource "azurerm_key_vault_secret" "POSTGRES-USER" {
```
Comment (timja) [inline infrastructure/main.tf]: probably not?
POSTGRES-USER

Comment (lgonczar) [reply infrastructure/main.tf]: changing it now!

File: infrastructure/main.tf [hunk 3]
```diff
@@ -7,7 +7,38 @@ resource "azurerm_resource_group" "rg" {
 }
 
 locals {
-  ase_name = "${data.terraform_remote_state.core_apps_compute.ase_name[0]}"
+  ase_name               = "${data.terraform_remote_state.core_apps_compute.ase_name[0]}"
+  is_preview             = "${(var.env == "preview" || var.env == "spreview")}"
+  previewVaultName       = "${var.product}-bsp"
```
Comment (rkondratowicz) [inline infrastructure/main.tf]: could this have a `-preview` prefix for clarity?

Comment (doncem) [reply infrastructure/main.tf]: I think it's following the current bulk scan processing setup - config doesn't have `-preview` suffix.

Comment (lgonczar) [reply infrastructure/main.tf]: As for `-preview` suffix, the name has to be the name of the existing vault (shared with bulk-scan-processor), so we can't change it just here. Yes, there's a plan to add common infrastructure (I'll add a story for it), but it's not a part of this one.

**Comment by lgonczar** [comment]
@doncem I agree, it should be specific to `bulk-scan` product, not any of its component. That should be done as part of this common component story.

**Comment by lgonczar** [comment]
Currently, can't run on preview, until common infrastructure definition is prepared. This project relies on what's created by bulk-scan-processor, which won't exist in preview. There's a story for that: https://tools.hmcts.net/jira/browse/RPE-680

**Pull request merged**